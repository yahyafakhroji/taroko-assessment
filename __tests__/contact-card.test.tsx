import 'node_modules/@testing-library/jest-dom';

import ContactCard from '@components/contact-card/contact-card.component';
import type { ContactListModel } from '@interfaces/contact.interface';
import { render } from 'node_modules/@testing-library/react';
import React from 'react';

describe('Card component', () => {
  it('renders correctly with data', () => {
    const data: ContactListModel = {
      id: 1,
      full_name: 'John Doe',
      first_name: 'John',
      last_name: 'John',
      job: 'Software Engineer',
      description: 'Lorem ipsum dolor sit amet',
      is_favorite: true,
    };
    const wrapper = render(<ContactCard data={data} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without data', () => {
    const wrapper = render(<ContactCard />);

    expect(wrapper).toMatchSnapshot();
  });
});
