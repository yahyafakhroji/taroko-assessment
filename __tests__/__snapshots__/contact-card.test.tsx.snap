// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly with data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="head"
        >
          <div
            class="profile"
          >
            <span
              class="avatar"
            >
              JD
            </span>
            <p>
              <span
                class="name"
              >
                John Doe
              </span>
              <span
                class="job"
              >
                Software Engineer
              </span>
            </p>
          </div>
          <div
            class="favorite active"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 576 512"
              width="1em"
            >
              <path
                d="M316.9 18c-5.3-11-16.5-18-28.8-18s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329l-24.6 145.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329l104.2-103.1c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7l-143.7-21.2L316.9 18z"
              />
            </svg>
          </div>
        </div>
        <div
          class="content"
        >
          Lorem ipsum dolor sit amet
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="head"
      >
        <div
          class="profile"
        >
          <span
            class="avatar"
          >
            JD
          </span>
          <p>
            <span
              class="name"
            >
              John Doe
            </span>
            <span
              class="job"
            >
              Software Engineer
            </span>
          </p>
        </div>
        <div
          class="favorite active"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M316.9 18c-5.3-11-16.5-18-28.8-18s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329l-24.6 145.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329l104.2-103.1c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7l-143.7-21.2L316.9 18z"
            />
          </svg>
        </div>
      </div>
      <div
        class="content"
      >
        Lorem ipsum dolor sit amet
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card component renders correctly without data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="head"
        >
          <div
            class="profile"
          >
            <span
              class="avatar"
            />
            <p>
              <span
                class="name"
              />
              <span
                class="job"
              />
            </p>
          </div>
          <div
            class="favorite undefined"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M6.1 21.98a1 1 0 01-1.45-1.06l1.03-6.03-4.38-4.26a1 1 0 01.56-1.71l6.05-.88 2.7-5.48a1 1 0 011.8 0l2.7 5.48 6.06.88a1 1 0 01.55 1.7l-4.38 4.27 1.04 6.03a1 1 0 01-1.46 1.06l-5.4-2.85-5.42 2.85zm4.95-4.87a1 1 0 01.93 0l4.08 2.15-.78-4.55a1 1 0 01.29-.88l3.3-3.22-4.56-.67a1 1 0 01-.76-.54l-2.04-4.14L9.47 9.4a1 1 0 01-.75.54l-4.57.67 3.3 3.22a1 1 0 01.3.88l-.79 4.55 4.09-2.15z"
              />
            </svg>
          </div>
        </div>
        <div
          class="content"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="head"
      >
        <div
          class="profile"
        >
          <span
            class="avatar"
          />
          <p>
            <span
              class="name"
            />
            <span
              class="job"
            />
          </p>
        </div>
        <div
          class="favorite undefined"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M6.1 21.98a1 1 0 01-1.45-1.06l1.03-6.03-4.38-4.26a1 1 0 01.56-1.71l6.05-.88 2.7-5.48a1 1 0 011.8 0l2.7 5.48 6.06.88a1 1 0 01.55 1.7l-4.38 4.27 1.04 6.03a1 1 0 01-1.46 1.06l-5.4-2.85-5.42 2.85zm4.95-4.87a1 1 0 01.93 0l4.08 2.15-.78-4.55a1 1 0 01.29-.88l3.3-3.22-4.56-.67a1 1 0 01-.76-.54l-2.04-4.14L9.47 9.4a1 1 0 01-.75.54l-4.57.67 3.3 3.22a1 1 0 01.3.88l-.79 4.55 4.09-2.15z"
            />
          </svg>
        </div>
      </div>
      <div
        class="content"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
